WINC-896, WINC-1162: Pick up openshift/kubernetes 1.29 rebase updates

jrvaldes opened the pull request:
Picks up openshift/kubernetes 1.29 rebase updates

jrvaldes committed:
[vendor] Use 1.29.0 k8s and OpenShift libraries

Ran:
go get k8s.io/api@v0.29.0
go get k8s.io/client-go@v0.29.0
go get k8s.io/cloud-provider@v0.29.0
go get k8s.io/kubectl@v0.29.0
go get k8s.io/kubelet@v0.29.0
go get -u github.com/openshift/api
go get -u github.com/openshift/client-go
go get -u github.com/openshift/library-go
go get -u github.com/openshift/machine-config-operator@e0dda8272f54efbf738ef43ea4b0e7b21450b7bf
go mod tidy
go mod vendor

jrvaldes committed:
[submodule][kubelet] Update to 1de6a149075

Update to https://github.com/openshift/kubernetes/commit/1de6a149075761f852d856f48dfb3663a6575751

This commit was generated using hack/update_submodules.sh

jrvaldes committed:
[build] Update kubelet version to v1.29.0+1de6a14

This commit was generated using hack/update_submodules.sh

jrvaldes committed:
[manifests] Set v1.29 as minimum k8s version

jrvaldes committed:
[e2e] Use VolumeResourceRequirements in kube API

ResourceRequirements struct was replaced with VolumeResourceRequirements for
use with volumes. https://github.com/kubernetes/kubernetes/pull/118653

jrvaldes commented:
/test aws-e2e-operator

jrvaldes commented:
/jira refresh

jrvaldes commented:
/test aws-e2e-operator

jrvaldes committed:
[nodeconfig] Adjust kubelet conf in unit test

Add ImageMaximumGCAge with default value to kubelet conf
in the unit test spec.
See https://github.com/kubernetes/kubernetes/pull/121275

jrvaldes commented:
/test aws-e2e-operator

jrvaldes commented:
aws-e2e-operator job [passed](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/2000/pull-ci-openshift-windows-machine-config-operator-master-aws-e2e-operator/1743034005793542144)

jrvaldes commented:
/jira refresh

alinaryan commented:
/lgtm

mansikulkarni96 commented:
/test aws-e2e-operator

jrvaldes commented:
> /test aws-e2e-operator

@mansikulkarni96  this test passed before. See https://github.com/openshift/windows-machine-config-operator/pull/2000#issuecomment-1881013354

I'm awaiting for reviews (approval) to trigger the full test suite.

jrvaldes commented:
/test remaining-required


to gather ci signal across platforms

sebsoto commented on pkg/nodeconfig/nodeconfig_test.go:
```diff
@@ -82,7 +82,7 @@ func TestCreateKubeletConf(t *testing.T) {
 			name:         "valid cidr",
 			cidr:         "10.0.128.8/24",
-			expectedSpec: "{\"kind\":\"KubeletConfiguration\",\"apiVersion\":\"kubelet.config.k8s.io/v1beta1\",\"syncFrequency\":\"0s\",\"fileCheckFrequency\":\"0s\",\"httpCheckFrequency\":\"0s\",\"rotateCertificates\":true,\"serverTLSBootstrap\":true,\"authentication\":{\"x509\":{\"clientCAFile\":\"C:\\\\k\\\\kubelet-ca.crt\"},\"webhook\":{\"cacheTTL\":\"0s\"},\"anonymous\":{\"enabled\":false}},\"authori ...
+			expectedSpec: "{\"kind\":\"KubeletConfiguration\",\"apiVersion\":\"kubelet.config.k8s.io/v1beta1\",\"syncFrequency\":\"0s\",\"fileCheckFrequency\":\"0s\",\"httpCheckFrequency\":\"0s\",\"rotateCertificates\":true,\"serverTLSBootstrap\":true,\"authentication\":{\"x509\":{\"clientCAFile\":\"C:\\\\k\\\\kubelet-ca.crt\"},\"webhook\":{\"cacheTTL\":\"0s\"},\"anonymous\":{\"enabled\":false}},\"authori ...
```
This is only changing the test not the Kubelet configuration

sebsoto replied:
Also the commit title says kubeconfig, when it should be kubelet config

jrvaldes replied:
Addressed.

sebsoto replied:
This is still not doing what is described in the commit title
`Add ImageMaximumGCAge to kubelet conf `
The conf isnt being changed. Is that meant to happen or is this only a unit test change?

jrvaldes replied:
this is only a unit test change, reworded the commit message

sebsoto reviewed (changes requested):
Thanks @jrvaldes mostly LGTM

mansikulkarni96 commented:
/approve

sebsoto commented:
/lgtm

jrvaldes commented:
/test remaining-required

saifshaikh48 reviewed (commented):
/lgtm

jrvaldes commented:
/retest-required

jrvaldes committed:
[deamon] Fix log format with wrong type

jrvaldes commented:
/test wicd-unit-vsphere

jrvaldes commented:
/test images

jrvaldes commented:
/test ci-bundle-wmco-bundle

jrvaldes commented:
/test vsphere-e2e-operator

jrvaldes commented:
/test images

saifshaikh48 commented:
/lgtm

jrvaldes commented:
/test remaining-required

jrvaldes commented:
/retest-required

mansikulkarni96 commented:
@jrvaldes the azure-e2e-operator failure needs to be investigated.

jrvaldes commented:
/hold

to investigate [issues](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_windows-machine-config-operator/2000/pull-ci-openshift-windows-machine-config-operator-master-azure-e2e-operator/1745259893402636288) with node creation in azure test

jrvaldes commented:
/test azure-e2e-upgrade

jrvaldes committed:
[services]: Always set node-ip for kubelet

This commit adjust the kubelet arguments to always
set the --node-ip flag with an initial value that
may be overrided by an external cloud provided. See https://github.com/kubernetes/kubernetes/pull/121028

Fixes WINC-896

jrvaldes commented:
/test remaining-required

jrvaldes committed:
[e2e] Remove NodeIPArg test

The --node-ip flag is set in kubelet arg for all instances, machine and
byoh, on all platforms, making this test irrelevant.

jrvaldes commented:
/test gcp-e2e-operator

jrvaldes commented:
/test azure-e2e-operator

jrvaldes commented:
/test azure-e2e-operator

jrvaldes commented:
/test remaining-required

to gather ci signal across cloud providers

jrvaldes commented:
/test azure-e2e-operator 

ipi-install failed

jrvaldes commented:
/test azure-e2e-operator

ipi-install failed

saifshaikh48 commented:
/lgtm

jrvaldes commented:
/hold cancel

ci is looking greener with https://github.com/openshift/windows-machine-config-operator/pull/2000/commits/629fd94635a22cf044a049a0dc760b7819c245bc

jrvaldes commented:
/jira refresh

jrvaldes commented:
/jira refresh
